wazo_webhookd/services/mobile/plugin.py

```diff
@@ -288,51 +300,73 @@ def __init__(
         self.jwt = jwt
 
     def cancelIncomingCall(self, data: dict[str, Any]) -> NotificationSentStatusDict:
+        payload = data | {'notification_time': generate_timestamp()}
         return self.send_notification(
             NotificationType.CANCEL_INCOMING_CALL,
             None,  # Message title
             None,  # Message body
-            {'items': data},
+            {'items': payload},
         )
 
     def incomingCall(self, data: dict[str, Any]) -> NotificationSentStatusDict:
+        payload = data | {'notification_time': generate_timestamp()}
+
+        if 'creation_time' not in payload:
+            logger.error(
+                'Missing creation_time in incoming call push notification event payload: %s',
+                data,
+            )
+        else:
+            try:
+                datetime.fromisoformat(payload['creation_time'])
+            except Exception:
+                logger.exception(
+                    'Invalid creation_time "%s" in incoming call '
+                    'push notification event payload: %s',
+                    payload['creation_time'],
+                    payload,
+                )
+                payload.pop('creation_time')
+
+        payload.setdefault('creation_time', payload['notification_time'])
```

Comment:
This is a lot of logic to maintain for an invalid event. We can trust events coming from the event bus. I would prefer simply passing the event along and skip the validation.